src/ucp/wireup/select.c

```diff
@@ -2384,9 +2385,8 @@ ucp_wireup_construct_lanes(const ucp_wireup_select_params_t *select_params,
             ucs_assert(key->am_lane == UCP_NULL_LANE);
             key->am_lane = lane;
         }
-        if ((select_ctx->lane_descs[lane].lane_types & UCS_BIT(UCP_LANE_TYPE_AM_BW)) &&
-            (lane < UCP_MAX_LANES - 1)) {
-            key->am_bw_lanes[lane + 1] = lane;
+        if ((select_ctx->lane_descs[lane].lane_types & UCS_BIT(UCP_LANE_TYPE_AM_BW))) {
+            key->am_bw_lanes[lane] = lane;
```

Comment:
what if am lane is not selected also as `AM_BW` lane? it will not be set in this array then

Reply:
With the new logic (last commit) AM lane is never selected as AM_BW since it passed as excluded lane.

Reply:
why do you start with index 0 then if you overwrite it with pure AM lane below

Reply:
If I understand the code correctly, it would never start with 0 since 0 would always be `am_lane` and `am_lane` cannot contain `UCP_LANE_TYPE_AM_BW` since it passed as `excl_lane` parameter during `add_bw_lanes()` call